dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1418,9 +1416,24 @@ void DeltaMergeStore::read(
     };
 
     GET_METRIC(tiflash_storage_read_tasks_count).Increment(tasks.size());
-    size_t final_num_stream
-        = enable_read_thread ? std::max(1, num_streams) : std::max(1, std::min(num_streams, tasks.size()));
```

Comment:
If `enable_read_thread` is enabled, in the old code, `final_num_stream` should be equal to `num_streams` here. Does this mean that when there are many small partitions, `num_streams` is passed in as `1`?

Reply:
see https://github.com/pingcap/tiflash/pull/10489#discussion_r2450971477